Update to core22, fuse-3.17.1, and sshfs-3.7.3

levkropp opened the pull request:
This PR modernizes the multipass-sshfs snap package with several important updates:

Changes
- Updated base from 'bare' to 'core22' (Ubuntu 22.04 LTS)
- Removed deprecated 'build-base: core18' configuration as it's not needed with core22
- Updated libfuse from 3.9.0 to 3.17.1
- Updated sshfs from 3.7.0 to 3.7.3
- Removed deprecated 'adapter: none' configuration which is no longer supported in newer snapcraft versions
- Added meson and ninja-build as explicit build dependencies for both libfuse and sshfs parts

Testing

The updated snap was tested with multipass 1.15.1 using the following process:

Checkout the `update-multipass-sshfs`
Build the snap locally
`snapcraft`
Create a test Multipass instance:
`multipass launch --name test-sshfs`
Transfer and installed the snap in the instance:
`multipass transfer multipass-sshfs_1.1-dirty_amd64.snap test-sshfs:`
`multipass exec test-sshfs -- sudo snap install --dangerous multipass-sshfs_1.1-dirty_amd64.snap`
Verify mount functionality:
```
# Create test directory and mount it
mkdir test_mount
multipass mount test_mount test-sshfs:/home/ubuntu/test_mount

# Test file creation and visibility
echo "test file" > test_mount/test.txt
multipass exec test-sshfs -- cat /home/ubuntu/test_mount/test.txt
```

ricab commented:
This is in a repo that we don't often monitor, so asking reviewers now to avoid loosing sight of it.

georgeliao commented:
About the build instruction: 

> Build the snap locally
> snapcraft

Does this allude that the `multipass-sshfs` repo is cloned and switched to `update-multipass-sshfs` branch? 
Apart from that, nice workflow for testing the repo, maybe adding a readme? Besides that, The code change looks sane to me.

xmkg commented:
Hey @levkropp, the changes look good to me. 

I'm just curious: Did you try using sshfs with the changes in place? One way to test that would be to build the multipass-sshfs snap, then copy & install the snap to the VM *before* mounting an sshfs mount (because mounting triggers the installation of the multipass-sshfs snap from the store).

levkropp commented:
> Does this allude that the `multipass-sshfs` repo is cloned and switched to `update-multipass-sshfs` branch?

Yes, sorry for the confusion I have updated the instructions reflecting this


> Did you try using sshfs with the changes in place? One way to test that would be to build the multipass-sshfs snap, then copy & install the snap to the VM _before_ mounting an sshfs mount (because mounting triggers the installation of the multipass-sshfs snap from the store).

@xmkg It seems that the multipass-sshfs snap doesn't get overwritten from the store, and mounts work using the updated multipass-sshfs snap.

```
frwk2@frwk2-laptop:~$ echo "test file" > test_mount/test.txt
multipass exec test-sshfs -- cat /home/ubuntu/test_mount/test.txt
test file
frwk2@frwk2-laptop:~$ multipass shell test-sshfs
Welcome to Ubuntu 24.04.2 LTS (GNU/Linux 6.8.0-57-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/pro

 System information as of Mon Apr 28 11:14:14 EDT 2025

  System load:  0.01              Processes:             104
  Usage of /:   50.4% of 3.80GB   Users logged in:       0
  Memory usage: 21%               IPv4 address for ens3: 10.190.155.146
  Swap usage:   0%


Expanded Security Maintenance for Applications is not enabled.

48 updates can be applied immediately.
40 of these updates are standard security updates.
To see these additional updates run: apt list --upgradable

Enable ESM Apps to receive additional future security updates.
See https://ubuntu.com/esm or run: sudo pro status


Last login: Mon Apr 28 11:14:24 2025 from 10.190.155.1
ubuntu@test-sshfs:~$ snap list multipass-sshfs
Name             Version    Rev  Tracking  Publisher  Notes
multipass-sshfs  1.1-dirty  x1   -         -          -
ubuntu@test-sshfs:~$
```
